packages/smooth_app/lib/pages/offline_data_page.dart

```diff
@@ -0,0 +1,223 @@
+                  ),
+                ),
+              ),
+              SliverList(
+                delegate: SliverChildListDelegate(
+                  <Widget>[
+                    UserPreferencesListTile(
+                      title: const Text(
+                        'Offline Product Data',
+                      ),
+                      subtitle: FutureBuilder<int>(
+                        future: daoProduct.getLength(),
+                        builder: (BuildContext context,
+                            AsyncSnapshot<int> snapshot) {
+                          if (snapshot.hasData) {
+                            return Text(
+                              '${snapshot.data} products available for immediate scanning',
+                            );
+                          } else {
+                            return Text(
+                                '$_length products available for immediate scanning');
+                          }
+                        },
+                      ),
+                      trailing: FutureBuilder<double>(
+                        future: daoProduct.getSize(),
+                        builder: (BuildContext context,
+                            AsyncSnapshot<double> snapshot) {
+                          if (snapshot.hasData) {
+                            return Text(
+                              '${snapshot.data} MB',
+                            );
+                          } else {
+                            return const CircularProgressIndicator();
+                          }
+                        },
+                      ),
+                    ),
+                    UserPreferencesListTile(
+                      trailing: const Icon(Icons.refresh),
+                      onTap: () async {
+                        final String status = await LoadingDialog.run<String>(
+                              context: context,
+                              future: _refreshLocalDatabase(daoProduct),
+                              title: 'Refreshing \n This may take a while',
+                              dismissible: true,
+                            ) ??
+                            'Refresh Cancelled';
+                        // TODO(ashaman999): dapproductlist.updateTimestamp();
+                        // ignore: use_build_context_synchronously
+                        ScaffoldMessenger.of(context).showSnackBar(
+                          SnackBar(
+                            content: Text(status),
+                          ),
+                        );
+                      },
+                      title: const Text(
+                        'Update Offline Data',
+                      ),
+                      subtitle: const Text(
+                        'Update the databse with the latest data from the server',
+                      ),
+                    ),
+                    UserPreferencesListTile(
+                      trailing: const Icon(Icons.delete),
```

Comment:
I'm very puzzled by this "delete" action, which is precisely the button no one should want to press. At least, show a "confirm" dialog.

Reply:
I was thinking about the same , hopefully you reminded me , i will add it for sure